### PR TITLE
Merge JIT/Methodical tests into the merged runners

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This PR merges many of the tests in the JIT/Methodical tree into their respective merged runners. This was done using the following process:

- Mechanically convert `<CLRTestPriority>1</CLRTestPriority>` to `[Outerloop]`.
- Mechanically convert `<GCStressIncompatible>true</GCStressIncompatible>` to `[SkipOnCoreClr(RuntimeTestMode.AnyGCStress)]`
- Mechanically convert `<JitOptimizationSensitive>true</JitOptimizationSensitive>` to `[SkipOnCoreClr(RuntimeTestMode.AnyJitStress)]`
- Mechanically convert `<HeapVerifyIncompatible>true</HeapVerifyIncompatible>` to `[SkipOnCoreClr(RuntimeTestMode.HeapVerify)]`
- Remove any `<RequiresProcessIsolation>true</RequiresProcessIsolation>` that was only there to support the above properties.
- Move any `<Compile />` entries for tests that are not `<RequiresProcessIsolation>true</RequiresProcessIsolation>` into the corresponding merged runner.
- Delete any now-unused `.csproj` projects.

A few tests were left with their csproj files intact as they include the same test source with different type definitions to reuse test code and merging them into the same assembly causes significant test size increase (ie. 30k line delta, +15k lines)

With this change, we remove ~1100 csproj files from the test tree (of 6982 csproj files today).

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
This cuts about a minute off of the managed test builds.

**Comment by jkoritzinsky** [comment]
Removing the extra test group used for building in Pri1, this now saves 4 minutes in a Pri1 build (comparing the run in the PR and a run against main).

File: src/tests/JIT/Methodical/AsgOp/i4/i4.cs [hunk 1]
```diff
@@ -670,7 +670,8 @@ private static int f109(int x, int y)
 
 
     [Fact]
-    public static int TestEntryPoint()
+        [OuterLoop]
```
Comment (ivdiazsa) [inline src/tests/JIT/Methodical/AsgOp/i4/i4.cs:673]: The indentation got a bit messed up here

Comment (jkoritzinsky) [reply src/tests/JIT/Methodical/AsgOp/i4/i4.cs:673]: Yeah some of these tests had really weird indentation to start with. I'll see if I can normalize it.

**Comment by ivdiazsa** [comment]
From what I'm seeing here, plus that the pipelines came back clean, it looks good to me! I think we should write a doc explaining how the new tags you added like `[Outerloop]` work, and whatnot. To make the process of adding new tests in the future easier :slightly_smiling_face:

**Review by ivdiazsa — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
/ba-g timeouts unrelated (the mono interpreter leg timed out before any tests even started to run)

**Pull request merged**